### ISSUE TITLE
Multiple panes cause incorrect errors

### ISSUE THREAD

**Issue by jdolle** [AtomLinter/linter-tslint#180]
When working with multiple panes in Atom, tslint will give invalid errors for files and behave erratically.

**Comment by Xapphire13** [comment]
Can you define _"behave erratically"_? Also, could you try with a different type of file/linter combo to see if the issue is actually [linter-ui-default](https://github.com/steelbrain/linter-ui-default)?

**Comment by roddypratt** [comment]
I've also seen cases where the linter shows spurious errors when using multiple panes.  Demo here (sorry about crap audio) https://screencast-o-matic.com/watch/cb1o2eXRGd

Doesn't seem to happen with Atom PHP linter.

**Comment by jdolle** [comment]
Sorry for the delay in responding -- I am running several linters and have not had issues with any other than tslint. It gives false errors and the error messages show incorrect values, i.e. instead of "property 'example' does not exist on type 'MyExample'", it may read "property 'ample Ev' does not exist on type 'ple And'". It is as if the linter loses its place in the file. If you need a screenshot I'll try to reproduce when I can find time.

**Comment by Xapphire13** [comment]
@roddypratt The video you shared shows the provider is `Typescript`, which is [this](https://github.com/TypeStrong/atom-typescript) package. I have also been having this issue with the TypeScript package lately (much worse issues in fact) but haven't filled a bug. Maybe you could share your video in a bug report on that repo? For reference, the provider from this package will look like this:
![image](https://user-images.githubusercontent.com/3392349/27206947-c3f866e4-51ef-11e7-8cb5-f4d467a2a12b.png)

@jdolle What does the provider field say for you?

**Comment by roddypratt** [comment]
@Xapphire13  You're right, of course. I'm new to Atom and hadn't sussed the distinction between the two plugins. I'll move the report and video over. Thanks!

**Comment by jdolle** [comment]
@Xapphire13 TSLint. It happened again to my coworker so I got screenshots.

This is with two panes open.
![broken_multi_pane](https://user-images.githubusercontent.com/1841898/27265299-93e8fb6c-5447-11e7-9cce-246449a9fb91.png)

And this is after restarting Atom.
![after_restarting_atom](https://user-images.githubusercontent.com/1841898/27265300-93e970e2-5447-11e7-8d17-ff1eca8210b6.png)

**Comment by Xapphire13** [comment]
I've been seeing this a little bit recently, but not isolated to linter-tslint (seen it with [atom-typescript](https://github.com/TypeStrong/atom-typescript) also). I think this _may_ be an issue with [linter-ui](https://github.com/steelbrain/linter-ui-default)

**Comment by mehcode** [comment]
Just to rule out the linter-ui, go ahead and test using [linter-ui-plus](https://atom.io/packages/linter-ui-plus) to see if you have the same problem. If you have the same problem it's likely not the linter-ui.

**Comment by Xapphire13** [comment]
I personally don't have a constant repro, @jdolle, is yours 100% repro?

**Comment by Xapphire13** [comment]
One interesting thing to note is that the lint issue list seems to be correct each time, just the in-editor indicators

**Comment by mehcode** [comment]
Oh. That actually explains a great deal. How are you seeing the problem? Is it after some plugin rewrites a bunch of a text in the editor?

**Comment by mehcode** [comment]
https://github.com/prettier/prettier-atom/commit/9a538e135a8c9e8c356ae736964815cbc6317984 

This might be related.

**Comment by Xapphire13** [comment]
I see the issue when i bulk delete a number of lines from one pane, then the atom-typescript package triggers static analysis, once that's done, the analysis results are messed up in one of the panes.

**Comment by Arcanemagus** [comment]
Hmmm, so the same file is open in multiple panes, editor markers are correct for the "active" pane, but the other pane the highlighting is wrong?

**Comment by Xapphire13** [comment]
As I recall (from my repros), yes. Though next time it happens for me I will take screenshots of my own so it's not just from memory =]

**Comment by jdolle** [comment]
@Xapphire13 I cannot reproduce it consistently. And I also get similar incorrect errors sometimes from Atom-Typescript so it may be an issue with the linter UI. The inconsistency makes it hard to know. Sorry I'm not much help.

**Comment by mxchange** [comment]
This is still an issue, I'm using `atom-typescript`, `linter-tslint` & `linter-ui-default`. 
It's hard to reproduce, but seems to have to do with multiple panes, and only restarting Atom will fix it.
Most times, a type error will occur, and when fixed, will still show red underlines, and often create new errors which are incorrect errors, on the wrong line, referring to other lines somehow.

**Comment by mxchange** [comment]
Example: 

![image](https://user-images.githubusercontent.com/24876534/29861010-1f7f11e4-8d68-11e7-9eda-3a3ea9fb1160.png)

**Comment by Xapphire13** [comment]
At this point im convinced the issue is **not** `linter-tslint` since most of the reports show this is happening for the `atom-typescript` package. I think the real issue is within `linter-ui-default` and have an open bug [here](https://github.com/steelbrain/linter-ui-default/issues/381).

Could people use that bug to add details from this point?
